### ISSUE TITLE
truncate_with_folder_marker shorten strategy does not work in $HOME

### ISSUE THREAD

**Issue by CedricFinance** [bhilburn/powerlevel9k#1004]
When the `truncate_with_folder_marker` stategy is used and `POWERLEVEL9K_DIR_PATH_ABSOLUTE` is not enabled, the dir segment is broken.

When I go to `$HOME/.config`, the prompt is `~~/.config`.

This is because of the `${current_path#${last_marked_folder}*}` at the end of the `truncate_with_folder_marker` case (powerlevel9k.zsh-theme:889). `current_path` is `~/.config` and `last_marked_folder` is `/home/myuser/.config`.

**Comment by dritter** [comment]
Thx for the report @CedricFinance .
For me it is even worse:
![bildschirmfoto 2018-11-04 um 19 47 31](https://user-images.githubusercontent.com/1544760/47968457-970a7400-e06a-11e8-8cdf-b93bb4b413d3.png)

Could you try #893 and see if that helps in your case? Planning to merge this PR in `next` branch soon..

**Comment by CedricFinance** [comment]
It's a bit better with #893 (the name of the folder containing the marker is missing).

<img width="534" alt="powerline9k_prompt_bug" src="https://user-images.githubusercontent.com/3935032/47986739-e3db6280-e0dd-11e8-9ed1-df09b02f701c.png">

You can take the unit test I've added in #1005 to make sure that the prompt is correct.

**Comment by dritter** [comment]
Thanks for the feedback. I improved #893 now and incorporated your Test. Could you try again?
Btw. I rebased #893 onto `next` branch, so you'll need to port your configuration over (plus side is that you are already prepared for the next release ;) )

**Comment by CedricFinance** [comment]
Nice. It's better.
<img width="436" alt="powerline9k_prompt_bug_2" src="https://user-images.githubusercontent.com/3935032/48158558-25812e80-e2d3-11e8-8cc6-dd72dd7a33e7.png">
The path is not shortened when we are in the folder with the marker.

I have written a test to highlight the bug.
```
function testTruncateWithFolderMarkerInMarkedFolder() {
  typeset -a P9K_LEFT_PROMPT_ELEMENTS
  P9K_LEFT_PROMPT_ELEMENTS=(dir)
  local P9K_DIR_SHORTEN_STRATEGY="truncate_with_folder_marker"

  local BASEFOLDER=/tmp/powerlevel9k-test
  local FOLDER=$BASEFOLDER/1/12
  mkdir -p $FOLDER
  # Setup folder marker
  touch $FOLDER/.shorten_folder_marker
  cd $FOLDER
  assertEquals "%K{004} %F{000}/…/12 %k%F{004}%f " "$(__p9k_build_left_prompt)"

  cd -
  rm -fr $BASEFOLDER
}
```

**Comment by dritter** [comment]
Thanks for the test. I incorporated it iny PR, fixed it and already merged it into `next`. Could you test it again?

**Comment by dritter** [comment]
Fixed with #893 . If there is still a problem, feel free to reopen.

**Comment by CedricFinance** [comment]
It's working fine now. Thanks!
